lib/hsa/hc_profile.cpp

```diff
@@ -0,0 +1,40 @@
+#include "hc_profile.hpp"
+
+#ifdef CODEXL_PROFILING_ENABLED
+#include "CXLActivityLogger.h"
+#endif
+
+cxlProfiler::cxlProfiler() {
+#ifdef CODEXL_PROFILING_ENABLED
+    if (getenv("HCC_CODEXL_PROFILING")) {
+        enabled = true;
+        cxlMarker::enabled = true;
+        amdtInitializeActivityLogger();
+    }
+#endif
+}
+cxlProfiler::~cxlProfiler() {
+#ifdef CODEXL_PROFILING_ENABLED
+    if (enabled) {
+        amdtFinalizeActivityLogger();
+    }
+#endif
+}
+
+bool cxlMarker::enabled = false;
```

Comment:
A global enabled flag is used to not use getenv everything a marker is created.

The enabled flag is set when the profiler is initialized and every marker that is created after doesn't have to check if the env var is set or not.

We can remove the dependency of hc_profile from user applications by not adding it in hcc-config, but we can still profile the runtime if the env var is set.